src/clinical/clinical-service.ts

```diff
@@ -355,60 +386,76 @@ export const getValidRecordsPostSubmission = async (
 
   const validDonorIds = donorData
     .filter(donor => donor.schemaMetadata.isValid)
-    .map(({ donorId }) => donorId);
-
-  const clinicalErrors: ClinicalErrorsResponseRecord[] = [];
+    .map(({ donorId }) => donorId)
+    .filter(id => typeof id === 'number');
```

Comment:
Will `donorId` ever not be a number? 
If it's always a number no need to have an additional iteration with `filter`

Reply:
Apparently 🤷 Donor is typed with a `?`, so from Typescript's perspective no, donorId will not always be a number
I will dig into this a bit